### PR TITLE
fix(Core/Maps): Include collision height in underwater status calcula…

### LINKED ISSUE #2208: Mounts and water Breath [$25]
if going in water on mount when posible see timer "Breath", but visual character on mount not in underwarer. But if unmount can see what character in fact underwater. 
https://i2.imageban.ru/out/2019/08/14/9effe0656d3c63ea72fe2b54c3749110.jpg
https://i4.imageban.ru/out/2019/08/14/7d092f329d4a0240f35856278de97b22.jpg
https://i6.imageban.ru/out/2019/08/14/23f87772e95eb131d68bf80e2b9f71c6.jpg

in addition, need dive deep for activate "Breath":
https://i6.imageban.ru/out/2019/08/14/0f4e9a37f95cf395d3a7f547bbe594fe.jpg
https://i1.imageban.ru/out/2019/08/14/af907c2c7b8d85eeb0a92a0c2821f1e9.jpg

<bountysource-plugin>

---
There is a **[$25 open bounty](https://www.bountysource.com/issues/78728630-mounts-and-water-breath?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
…tions.
Fixed #2208.

## Changes Proposed:
-  
-  

## Issues Addressed:
- Closes #2208.

## Tests Performed:
-
-

## How to Test the Changes:

1. Mount
2. Enter water.
3. Breath timer is gone.

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@UltraNix thanks for this contribution, please fill out the testing information so people know how to test this

**Comment by UltraNix** [comment]
Done.
More info in addressed issue.

**Review by Yehonal — APPROVED** [review]

File: src/server/game/Maps/Map.h [hunk 1]
```diff
@@ -208,7 +208,7 @@ class GridMap
     [[nodiscard]] float getMinHeight(float x, float y) const;
     [[nodiscard]] float getLiquidLevel(float x, float y) const;
     [[nodiscard]] uint8 getTerrainType(float x, float y) const;
-    ZLiquidStatus getLiquidStatus(float x, float y, float z, uint8 ReqLiquidType, LiquidData* data = nullptr);
+    ZLiquidStatus getLiquidStatus(float x, float y, float z, uint8 ReqLiquidType, float collisionHeight, LiquidData* data = nullptr);
```
Comment (pklloveyou) [inline src/server/game/Maps/Map.h:211]: Does it need a default value?
```
ZLiquidStatus getLiquidStatus(float x, float y, float z, uint8 ReqLiquidType, LiquidData* data = nullptr, float collisionHeight = DEFAULT_COLLISION_HEIGHT);
```

Comment (UltraNix) [reply src/server/game/Maps/Map.h:211]: Nope, because it's called only within `Map::getLiquidStatus`, where you have already collisionHeight defined.

Comment (pklloveyou) [reply src/server/game/Maps/Map.h:211]: ok

**Comment by pklloveyou** [comment]
Tested

**Comment by UltraNix** [comment]
Updated.

**Review by Yehonal — APPROVED** [review]

**Pull request merged**